README.md

```diff
@@ -34,62 +31,73 @@ Note the the `default.nix` is a special name and thus you can use the path to be
 The functions it produces are as follows.
 
 ### `rustc`
 
 `rustc` is a function from a set of optional arguments to a derivation for the rust compiler.
-`date` determines which nightly to download (defaults to latest date where a nightly was successfully built).
-`system` is the system the compiler *runs* on (defaults to the system Nix is installed on).
-This is an LLVM- (and Rust-) style system name, not Nix-style.
-`hash` is the expect hash of the nightly download.
-The default here is somewhat complex, and marginally better than just an unchecked download.
-If no hash is provided, rust-nightly-nix will first download the hash from Mozilla (an unchecked download), then use the downloaded hash to verify the compiler download.
-That way, the compiler download is always verified with some hash.
-
-As an example example,
+
+* `date` determines which nightly to download; defaults to latest date where a nightly was successfully built.
+* `system` is the system the compiler *runs* on; defaults to the system Nix is installed on.
+  This is an LLVM- (and Rust-) style system name, not Nix-style.
+* `hash` is the expect hash of the nightly download.
+   If no hash is provided, rust-nightly-nix will first download the hash from Mozilla, which is an unchecked download, and then use the downloaded hash to verify the compiler download.
+   This way, the compiler download is always verified with some hash.
+
+#### Example Usage
+
+Build a nightly for June 9th, 2016.
+
 ```nix
 {
-  funs = callPackage /path/to/rust-nightly-nix { };
+  rustPackages = callPackage /path/to/rust-nightly-nix { };
 
-  rustcNightly = funs.rustc {
+  rustcNightly2016-06-09 = rustPackages.rustc {
     date = "2016-06-09";
     hash = "1p0xkpfk66jq0iladqfrhqk1zc1jr9n2v2lqyf7jjbrmqx2ja65i";
   };
 }
 ```
-provides a hash to add some security to the build process.
 
 ### `cargo`
 
-`cargo` has the same type as `rustc`.
-Note that if one of a nightly Rust compiler build and nightly Cargo build fails and the other succeeds, the other is still published.
-So don't freak out if on some day half your downloads are failing.
+`cargo` has the same imput as `rustc`, but returns a derivation for Cargo.
+
+Note that if either the Rust or Cargo nightly build fails, the other is still published.
+Don't freak out if on some days half of your downloads are failing.
 
 ### `rust-std`
 
-`rust-std` is used for prebuilt standard libraries.
-On "normal" platforms this is `std` and its dependencies but on more exotic platforms (e.g. small CPUs for embedded systems) it will contain less as `std` itself makes no sense.
-It has the same type as `rustc`.
+`rust-std` is used for pre-built standard libraries.
+
+`rust-std` has the same imput as `rustc`.
+
+On "normal" platforms this is `std` and its dependencies, but on more exotic platforms (e.g. small CPUs for embedded systems) it will contain less as `std` itself makes no sense.
```

Comment:
> it will contain less as `std` itself makes no sense.

I suggest:

> which cannot fully support `std` it will contain a subset of the standard library crates such as `core`.